dot/network/helpers_test.go

```diff
@@ -18,15 +19,19 @@ import (
 )
 
 type testStreamHandler struct {
+	sync.Mutex
```

Comment:
nit Maybe name the mutex so we know what it is protecting? Is it the `messages` map? Or something else?